modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/package.scala

```diff
@@ -201,9 +201,11 @@ package object engine {
       _.map { seq =>
         seq match {
           case Sequence.State.Final(_,_) => unit
-          case _          => Nondeterminism[Task].gatherUnordered (
-            seq.current.actions.zipWithIndex.map (act)
-            ).liftM[HandleStateT] *> send (q) (executed (id) )
+          case _                         => Task.delay {
+            (Nondeterminism[Task].gatherUnordered (
+              seq.current.actions.zipWithIndex.map (act)
+            ) *> q.enqueueOne(executed (id) )).unsafePerformAsync(x => ())
```

Comment:
I've been bitten several times by Tasks getting stuck that when using `unsafePerformAsync` made it work. In almost every case there was an extra Task wrapping the true Task that needs to be removed. The double Task can be hard to spot because the type checker can't catch it. Here, at a glance, I can't find where it is though.